tests/baselines/reference/errorOnUnionVsObjectShouldDeeplyDisambiguate.errors.txt

```diff
@@ -0,0 +1,77 @@
+tests/cases/compiler/errorOnUnionVsObjectShouldDeeplyDisambiguate.ts(17,7): error TS2322: Type '() => number[]' is not assignable to type '() => Promise<number[]>'.
+  Type 'number[]' is missing the following properties from type 'Promise<number[]>': then, catch

```

Comment:
As a separate change, it might be nice to special case the message `Foo is not assignable to Promise<Foo>` to say something about adding `await`, rather than listing missing properties.

Reply:
I don't think it's as much about missing `await` since it's the target - but https://github.com/microsoft/TypeScript/issues/35300 covers the basic idea here.